src/network/networkd-route.c

```diff
@@ -1399,32 +1399,58 @@ bool route_can_update(const Route *existing, const Route *requesting) {
         assert(existing);
         assert(requesting);
 
-        if (route_compare_func(existing, requesting) != 0)
+        if (route_compare_func(existing, requesting) != 0) {
+                log_route_debug(existing, "Cannot update route, as the existing route is different", existing->manager);
                 return false;
+        }
 
         switch (existing->family) {
         case AF_INET:
-                if (existing->nexthop.weight != requesting->nexthop.weight)
+                if (existing->nexthop.weight != requesting->nexthop.weight) {
+                        log_debug("existing weight: %u, requesting weight: %u",
+                                  existing->nexthop.weight, requesting->nexthop.weight);
                         return false;
+                }
                 return true;
 
         case AF_INET6:
-                if (existing->protocol != requesting->protocol)
+                if (existing->protocol != requesting->protocol) {
+                        log_debug("existing protocol: %hhu, requesting protocol: %hhu",
```

Comment:
Use `route_protocol_to_string_alloc()` here to print the string version of the protocol. Enclose in `if (DEBUG_LOGGING)` to only allocate if we're debug logging.

Reply:
`log_debug("Route protocol mismatch: %s != %s")`